apps/runtime-demo/host/webpack.config.js

```diff
@@ -2,11 +2,50 @@ const { registerPluginTSTranspiler } = require('nx/src/utils/nx-plugin.js');
 registerPluginTSTranspiler();
 
 const { composePlugins, withNx } = require('@nx/webpack');
+const { withReact } = require('@nx/react');
 
-module.exports = composePlugins(withNx(), (config) => {
-  // Update the webpack config as needed here.
-  // e.g. `config.plugins.push(new MyPlugin())`
-  config.devtool = false;
+const path = require('path');
+const baseConfig = require('./module-federation.config');
+const { ModuleFederationPlugin } = require('@module-federation/enhanced');
+/**
+ * @type {import('@nx/react/module-federation').ModuleFederationConfig}
+ **/
+const defaultConfig = {
+  ...baseConfig,
+};
 
-  return config;
-});
+module.exports = composePlugins(
+  withNx(),
+  withReact(),
+  async (config, context) => {
+    config.plugins.push(new ModuleFederationPlugin(defaultConfig));
+    config.plugins.forEach((p) => {
+      if (p.constructor.name === 'ModuleFederationPlugin') {
+        //Temporary workaround - https://github.com/nrwl/nx/issues/16983
+        p._options.library = undefined;
+      }
+    });
+
+    config.devServer = {
+      ...(config.devServer || {}),
+      //Needs to resolve static files from the dist folder (@mf-types)
```

Comment:
Wait does this mean we are getting TypeScript type sync in initial release?!

Reply:
Ah, sorry no, this is the configuration I copied from another demo. 

But typescript type sync can be added in next time , the waiting time shouldn't be long

Reply:
Okay no problem